### PR TITLE
[5.3.0][FIX] Prevent crash when funds warning is pressed

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
The app is crashing when we do not have enough funds to do a transaction on testnet

**Proposed Solution**
The app will redirect to the MetaMask faucet when it's testnet
The app will redirect to onRamp with the crypto we are trying to transact

**Code Impact**
Medium, was changed in all transaction confirmations screens/modals:
* Confirm screen of send flow
* Modal of transaction Review
* Modal of transaction approval 

**Test Cases**
(For mainnet, avax and rinkeby)
Case 1:
* Press the warning on the confirm screen of send flow
Case 2:
* Press the warning on transaction modal of deeplinks
Case 3:
* Press the warning on transaction modal of webview

**Screenshots/Recordings**
Bug: 
http://recordit.co/wpPgIUiPcG

Solution:
https://recordit.co/KR5KHCtYOV

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Progresses #4177

### CODE REVIEW

**Review by gantunesr — APPROVED** [review]
lgtm

**Review by cortisiko — CHANGES_REQUESTED** [review]
Some custom networks have the ability to onramp. for example BSC, Polygon, Avalanche.
With that being said if i were to be on any of those custom networks (in my case i was on polygon) & i tap the warning message I am taken the [faucet page](http://recordit.co/6aaVOnVq9g)

To reproduce:
- with an account with no funds add a custom network (for example BSC, Polygon, Avalanche.)
- go to https://metamask.github.io/test-dapp/
- connect your wallet
- tap on deploy contract, or create token
- now tap on the warning message
- notice the error

File: app/components/Views/SendFlow/Confirm/index.js [hunk 1]
```diff
@@ -1244,12 +1244,13 @@ class Confirm extends PureComponent {
...
 			);
 
-		const is_main_net = isMainNet(network);
-		const errorPress = is_main_net ? this.buyEth : this.gotoFaucet;
-		const networkName = capitalize(getNetworkName(network));
-		const errorLinkText = is_main_net
-			? strings('transaction.buy_more_eth')
-			: strings('transaction.get_ether', { networkName });
+		const is_test_net = isTestNet(network);
+		const errorPress = is_test_net
```
Comment (tommasini) [inline app/components/Views/SendFlow/Confirm/index.js:1248]: Need to think more about this, and do a deeper look into the navigation stack
Please feel free to comment another idea to navigate to the faucet if it is a testnet in the send flow confirm screen

File: app/util/networks/index.js [hunk 2]
```diff
@@ -69,6 +81,51 @@ const NetworkList = {
 		color: '#f2f3f4',
 		networkType: 'rpc',
 	},
+	[BSC]: {
```
Comment (tommasini) [inline app/util/networks/index.js:84]: This hexChainId are wrong, they need to be fixed

**Comment by tommasini** [comment]
* Replaced all the places where transaction needs to be confirmed 
* The message is according to the coin that you are trying to transact
* New coins added, BSC, Polygon, Avalanche, Celo and Fantom

https://images.zenhubusercontent.com/141427485/57f78fe9-f2e6-4858-9eee-d690f317477d/simulator_screen_recording___iphone_11_pro___2022_05_02_at_18_15_01.mp4

File: locales/languages/en.json [hunk 3]
```diff
@@ -1,1904 +1,1904 @@
 {
```
[the remark this answers is not in the document]
Comment (tommasini) [reply locales/languages/en.json:1]: Done

File: app/components/Views/SendFlow/Confirm/index.js [hunk 4]
```diff
@@ -1306,10 +1301,10 @@ class Confirm extends PureComponent {
   };
 
   gotoFaucet = () => {
```
[the remark this answers is not in the document]
Comment (tommasini) [reply app/components/Views/SendFlow/Confirm/index.js:1303]: Done

File: app/components/UI/TransactionReview/TransactionReviewInformation/index.js [hunk 5]
```diff
@@ -613,13 +608,12 @@ class TransactionReviewInformation extends PureComponent {
     const colors = this.context.colors || mockTheme.colors;
     const styles = createStyles(colors);
 
-    const is_main_net = isMainNet(network);
+    const is_test_net = isTestNet(network);
```
[the remark this answers is not in the document]
Comment (tommasini) [reply app/components/UI/TransactionReview/TransactionReviewInformation/index.js:611]: done

File: app/constants/network.js [hunk 6]
```diff
@@ -6,3 +6,8 @@ export const GOERLI = 'goerli';
 export const RPC = 'rpc';
 export const NO_RPC_BLOCK_EXPLORER = 'NO_BLOCK_EXPLORER';
 export const PRIVATENETWORK = 'PRIVATENETWORK';
+export const BSC = 'bsc';
+export const POLYGON = 'polygon';
+export const AVALANCHE = 'avalanche';
+export const CELO = 'celo';
+export const fantom = 'fantom';
```
[the remark this answers is not in the document]
Comment (tommasini) [reply app/constants/network.js:13]: Done

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮 🌮

File: app/util/networks/index.js [hunk 7]
```diff
@@ -101,6 +101,20 @@ export const isMainnetByChainId = (chainId) =>
 export const getNetworkName = (id) =>
   NetworkListKeys.find((key) => NetworkList[key].networkId === Number(id));
 
+export const isTestNet = (networkId) => {
+  const networkName = getNetworkName(networkId);
+
+  if (
```
Comment (Fatxx) [inline app/util/networks/index.js:107]: We can do instead `return networkName === ROPSTEN ||
    networkName === GOERLI ||
    networkName === KOVAN ||
    networkName === RINKEBY`

**Review by Fatxx — CHANGES_REQUESTED** [review]

**Review by Fatxx — APPROVED** [review]
LGTM

**Pull request merged**